silk-workbench/silk-workbench-workflow/app/controllers/workflowApi/WorkflowApi.scala

```diff
@@ -218,22 +220,191 @@ class WorkflowApi @Inject()() extends InjectedController with ControllerUtilsTra
+    val activity = workflowTask.activity[WorkflowWithPayloadExecutor]
+    val id = activity.start(workflowConfig)
+    val result = StartActivityResponse(activity.name, id)
+    Created(Json.toJson(result))
+  }
+
+  @Operation(
+    summary = "Parameterized workflow execution result (asynchronous)",
+    description = "Returns the result of a workflow execution.",
+    responses = Array(
+      new ApiResponse(
+        responseCode = "200",
+        description = "If the workflow was successful. The result of the variable output dataset is returned in the response."
+      ),
+      new ApiResponse(
+        responseCode = "204",
+        description = "If the workflow was successful, but contains no variable output dataset."
+      ),
+      new ApiResponse(
+        responseCode = "400",
+        description = "If the workflow has not completed successfully at this point."
+      )
+    )
+  )
+  def variableWorkflowAsyncResult(@Parameter(
+                                    name = "projectId",
+                                    description = "The project identifier",
+                                    required = true,
+                                    in = ParameterIn.PATH,
+                                    schema = new Schema(implementation = classOf[String])
+                                  )
+                                  projectName: String,
+                                  @Parameter(
+                                    name = "workflowId",
+                                    description = "The workflow identifier",
+                                    required = true,
+                                    in = ParameterIn.PATH,
+                                    schema = new Schema(implementation = classOf[String])
+                                  )
+                                  workflowTaskName: String,
+                                  @Parameter(
+                                    name = "instanceId",
+                                    description = "The activity instance identifier",
+                                    required = true,
+                                    in = ParameterIn.PATH,
+                                    schema = new Schema(implementation = classOf[String])
+                                  )
+                                  instanceId: String): Action[AnyContent] = RequestUserContextAction { implicit request => implicit userContext =>
+    val (project, workflowTask) = getProjectAndTask[Workflow](projectName, workflowTaskName)
+
+    // Make sure that the activity has been successful
+    val activity = workflowTask.activity[WorkflowWithPayloadExecutor].instance(instanceId)
+    if(activity.status().isRunning) {
+      throw BadUserInputException("Workflow is still running.")
+    } else if(activity.status().failed) {
+      throw BadUserInputException("Workflow failed.", activity.status().exception)
```

Comment:
Should this be rather a 5xx error instead of a 400 error? In the end it's not clear if the exception happened because of the user provided data or something internally.